tests/networkpolicy_test.go

```diff
@@ -15,28 +15,23 @@ import (
 	cd "kubevirt.io/kubevirt/tests/containerdisk"
 )
 
-func assertPingSucceed(ip string, vmi *v1.VirtualMachineInstance) {
-	expecter, err := tests.LoggedInCirrosExpecter(vmi)
-	Expect(err).ToNot(HaveOccurred())
-	defer expecter.Close()
-
-	err = tests.CheckForTextExpecter(vmi, []expect.Batcher{
-		&expect.BSnd{S: fmt.Sprintf("ping -w 3 %s \n", ip)},
-		&expect.BExp{R: "0% packet loss"},
-	}, 60)
-	Expect(err).ToNot(HaveOccurred())
+func pingEventually(fromVmi *v1.VirtualMachineInstance, toIp string) AsyncAssertion {
+	return Eventually(func() error {
+		By(fmt.Sprintf("Pinging from VMI %s/%s to Ip %s", fromVmi.Namespace, fromVmi.Name, toIp))
+		return tests.PingFromVMConsole(fromVmi, toIp)
+	}, 10*time.Second, time.Second)
 }
 
-func assertPingFail(ip string, vmi *v1.VirtualMachineInstance) {
-	expecter, err := tests.LoggedInCirrosExpecter(vmi)
-	Expect(err).ToNot(HaveOccurred())
-	defer expecter.Close()
+func assertPingSucceedBetweenVMs(vmisrc, vmidst *v1.VirtualMachineInstance) {
+	for _, ip := range vmidst.Status.Interfaces[0].IPs {
+		pingEventually(vmisrc, ip).Should(Succeed())
```

Comment:
Can we add the "should" string at the end of `Succeed()` ?

Reply:
Done